java/test/jmri/jmrix/lenz/XNetReplyTest.java

```diff
@@ -1387,7 +1395,288 @@ public void testToMonitorStringInvalidFeedbackReply(){
+        
+        selected = r.selectTurnoutFeedback(24);
+        assertTrue(selected.isPresent());
+        evenItem = selected.get();
+        assertTurnoutFeedbackData(r, 1, 24, 1, 0, evenItem);
+    }
+    
+    /**
+     * Checks that feedback module item gives invalid / erroneous / null
+     * information when used as accessory.
+     */
+    @Test
+    public void testOtherRepliesAsAccessoryFeedback() {
+        XNetReply r = new XNetReply("42 05 58 0f");
+        // test directly the item
+        FeedbackItem item = new FeedbackItem(r, 45, 0x58);
+        assertEquals(45, item.getAddress());
+        for (int a = 45; a < 45 + 4; a++) {
+            assertTrue(item.matchesAddress(a));
+            // last bit is set, all others are false.
+            assertEquals("Bit state for sensor " + a, a == 48, item.getEncoderStatus(a));
+        }
+        // does not match accessory for 0x05, 0x58
+        assertFalse(item.matchesAddress(21));
+        assertEquals(3, item.getAccessoryStatus());
+        assertNull(item.pairedAccessoryItem());
+
+        // check that no turnout feedback can be selected
+        for (int i = 1 ; i < 1024; i++) {
+            assertFalse(r.selectTurnoutFeedback(i).isPresent());
+        }
+        
+        // no accessory feedback present
+        r = new XNetReply("E3 40 C1 04 61");
+        for (int i = 1 ; i < 1024; i++) {
+            assertFalse(r.selectTurnoutFeedback(i).isPresent());
+        }
+    }
+    
+    /**
+     * Checks that encoder feedback will return null for turnout
+     * feedbacks.
+     */
+    @Test
+    public void testOtherRepliesAsEncoder() {
+        XNetReply r = new XNetReply("42 05 28 0f");
+        for (int i = 1 ; i < 1024; i++) {
+            assertNull(r.selectModuleFeedback(i));
+        }
+        r = new XNetReply("E3 40 C1 04 61");
+        for (int i = 1 ; i < 1024; i++) {
+            assertNull(r.selectModuleFeedback(i));
+        }
+
+    }
+
+    
+    /**
+     * Checks that select will filter out accessories with an incorrect state.
+     */
+    @Test
+    public void testInvalidAccessoryStateFiltered() {
+        XNetReply r = new XNetReply("42 05 2B 0f");
+        Optional<FeedbackItem> opt = r.selectTurnoutFeedback(21);
+        assertFalse(opt.isPresent());
+        assertEquals(-1, r.getTurnoutStatus(1));
+        
+        opt = r.selectTurnoutFeedback(22);
+        assertTrue(opt.isPresent());
+        assertEquals(Turnout.THROWN, opt.get().getTurnoutStatus());
+        assertEquals(r.getTurnoutStatus(0), opt.get().getTurnoutStatus());
+    }
+    
+    @Test
+    public void testSingleEncoderModuleFeedback() {
+        // feedback 5 * 8  + 4 (upper nibble) (+1) = 45
+        XNetReply r = new XNetReply("42 05 58 0f");
+        
+        assertNull(r.selectModuleFeedback(44));
+        
+        for (int i = 45; i < 45 + 4; i++) {
+            Boolean b = r.selectModuleFeedback(i);
+            assertNotNull(b);
```

Comment:
Not sure what it means to assertNonNull on a Boolean. Is that really what was intended? (Several places)

Reply:
Boolean objects actually can be null.  It becomes a tristate variable:  True,False, or not Defined.  One of my co-workers uses this somewhat frequently.

The method this is testing is returning null if the reply this is applied to is not a feedback message.  If it is a feedback message, it returns the true/false value of the appropriate bit in the feedback message.

So it is a little different, but the meaning is correct.

Reply:
I did scratch my head with this return type: `Optional<Boolean>` would be more API consistent, lambda / stream friendly etc etc. But seemed so ridiculous...